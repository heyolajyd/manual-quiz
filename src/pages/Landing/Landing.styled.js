import styled from 'styled-components';

export const Hero = styled.section`
  background: url(/assets/hair-cat-header.png) center;
  background-size: cover;
  min-height: 680px;
  padding: 30px 0 0 80px;

  article {
    max-width: 380px;
    margin: 22px 0 38px;

    p {
      font-size: ${(props) => props.theme.textSizes.large};
      font-family: ${(props) => props.theme.fonts.helvetica};
      line-height: 26px;
    }
  }
`;

export const HeroText = styled.h1`
  font-size: ${(props) => props.theme.textSizes.xLarge};
  color: ${(props) => props.theme.colors.tealDark};
  font-family: ${(props) => props.theme.fonts.montserrat};
  font-weight: ${props => props.theme.fontWeight.bold};
  padding-top: 6.5rem;
  text-transform: uppercase;
  max-width: 600px;
  margin-bottom: 0;
  line-height: 98px;
`;
