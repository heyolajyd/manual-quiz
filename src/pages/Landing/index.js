import React from 'react';
import PropTypes from 'prop-types';
import Button from 'components/Button';
import Logo from 'components/Logo';
import Footer from 'components/Footer';
import HairLoss from 'assets/hair-photo.png';
import SexPhoto from 'assets/sex-photo.png';
import {
  Hero,
  HeroText,
  ProductSection,
  ProductItem,
  NumberText,
  ProductDetails,
  ProductSpec,
  Text,
} from './Landing.styled';

const Landing = ({ startQuiz }) => {
  return (
    <main>
      <Hero>
        <Logo />
        <HeroText>
          <div>Be good</div>
          <div>to yourself</div>
        </HeroText>
        <article>
          <p>
            We’re working around the clock to bring you a holistic approach to
            your wellness. From top to bottom, inside and out.
          </p>
        </article>
        <Button onClick={startQuiz}>TAKE THE QUIZ</Button>
      </Hero>
      <ProductSection>
        <h2>What we can help with</h2>
        <ProductItem>
          <img src={HairLoss} alt='hair loss' />
          <ProductDetails>
            <NumberText>01</NumberText>
            <ProductSpec>
              <Text>Hair Loss</Text>
              <h5>Hair loss needn’t be irreversible. We can help!</h5>
              <p>
                Hair loss needn’t be irreversible. There’s a scientifically
                proven way that’s most effective in keeping and regrowing your
                hair. It’s all to do with blocking a pesky hormone called DHT.
                That’s the bad guy behind hair loss. And you can keep him at
                bay. The choice is yours.
              </p>
            </ProductSpec>
          </ProductDetails>
        </ProductItem>
        <ProductItem className='reversed'>
          <ProductDetails>
            <NumberText>02</NumberText>
            <ProductSpec className='reversed'>
              <Text>Erectile Dysfunction</Text>
              <h5>
                Erections can be a tricky thing. But no need to feel down!
              </h5>
              <p>
                There are plenty of reasons why you might be having difficulty
                in the erection department. We can help you figure out possible
                reasons why. And prescribe a pill if needed.
              </p>
            </ProductSpec>
          </ProductDetails>
          <img src={SexPhoto} alt='sex' />
        </ProductItem>
      </ProductSection>
      <Footer />
    </main>
  );
};

Landing.propTypes = {
  startQuiz: PropTypes.func,
};

export default Landing;
