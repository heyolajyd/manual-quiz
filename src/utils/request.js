/**
 * Util function to get questions
 */
export const getQuestion = async () => {
  try {
    const res = await fetch('./questions.json');
    return await res.json();
  } catch (err) {
    console.log('Failed to parse: ', JSON.stringify(err));
    throw new Error(err);
  }
};
