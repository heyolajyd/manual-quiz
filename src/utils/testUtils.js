import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from 'styles/global';

export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
