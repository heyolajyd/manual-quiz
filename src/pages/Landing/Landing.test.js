import React from 'react';
import { fireEvent } from '@testing-library/react';
import { withThemeProvider } from 'utils/testUtils';
import Landing from './index';

const startQuizMock = jest.fn();

describe('Landing page', function () {
  it('should match snapshot', () => {
    const { container } = withThemeProvider(<Landing />);

    expect(container).toMatchSnapshot();
  });

  it('should start quiz', () => {
    const { getByText } = withThemeProvider(
      <Landing startQuiz={startQuizMock} />
    );

    fireEvent.click(getByText(/TAKE THE QUIZ/));

    expect(startQuizMock).toBeCalled();
  });
});
