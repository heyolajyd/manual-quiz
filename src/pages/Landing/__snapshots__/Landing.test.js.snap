// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page should match snapshot 1`] = `
<div>
  <main>
    <section
      class="sc-fzoLsD epcALz"
    >
      <svg
        class="sc-AxirZ ktiAvJ"
      >
        manual.svg
      </svg>
      <h1
        class="sc-fzpans cQhUVb"
      >
        <div>
          Be good
        </div>
        <div>
          to yourself
        </div>
      </h1>
      <article>
        <p>
          We’re working around the clock to bring you a holistic approach to your wellness. From top to bottom, inside and out.
        </p>
      </article>
      <button
        class="sc-AxjAm ewqnpQ"
      >
        TAKE THE QUIZ
      </button>
    </section>
    <section
      class="sc-fzplWN kpDZxX"
    >
      <h2>
        What we can help with
      </h2>
      <div
        class="sc-fznyAO kGYtkF"
      >
        <img
          alt="hair-loss"
          src="hair-photo.png"
        />
        <div
          class="sc-fznKkj javTWE"
        >
          <span
            class="sc-fznZeY dtsngr"
          >
            01
          </span>
          <article
            class="sc-fzokOt EwiZm"
          >
            <h6
              class="sc-fzqBZW bpmzeA"
            >
              Hair Loss
            </h6>
            <h5>
              Hair loss needn’t be irreversible. We can help!
            </h5>
            <p>
              Hair loss needn’t be irreversible. There’s a scientifically proven way that’s most effective in keeping and regrowing your hair. It’s all to do with blocking a pesky hormone called DHT. That’s the bad guy behind hair loss. And you can keep him at bay. The choice is yours.
            </p>
          </article>
        </div>
      </div>
      <div
        class="sc-fznyAO kGYtkF reversed"
      >
        <div
          class="sc-fznKkj javTWE"
        >
          <span
            class="sc-fznZeY dtsngr"
          >
            02
          </span>
          <article
            class="sc-fzokOt EwiZm reversed"
          >
            <h6
              class="sc-fzqBZW bpmzeA"
            >
              Erectile Dysfunction
            </h6>
            <h5>
              Erections can be a tricky thing. But no need to feel down!
            </h5>
            <p>
              There are plenty of reasons why you might be having difficulty in the erection department. We can help you figure out possible reasons why. And prescribe a pill if needed.
            </p>
          </article>
        </div>
        <img
          alt="sex"
          src="sex-photo.png"
        />
      </div>
    </section>
    <footer
      class="sc-AxiKw dpzIdq"
    >
      <div
        class="row"
      >
        <div
          class="sc-AxhCb hAtPWR col-sm-3 col-12"
        >
          <svg
            class="sc-AxirZ bEopbj"
            height="75"
            width="80"
          >
            manual.svg
          </svg>
        </div>
        <div
          class="col-sm-2 col-4"
        >
          <h5
            class="sc-AxhUy kaKyLb"
          >
            Product
          </h5>
          <a
            class="sc-AxgMl gwITOl"
          >
            Popular
          </a>
          <a
            class="sc-AxgMl gwITOl"
          >
            Trending
          </a>
          <a
            class="sc-AxgMl gwITOl"
          >
            Guided
          </a>
          <a
            class="sc-AxgMl gwITOl"
          >
            Products
          </a>
        </div>
        <div
          class="col-sm-2 col-4"
        >
          <h5
            class="sc-AxhUy kaKyLb"
          >
            Company
          </h5>
          <a
            class="sc-AxgMl gwITOl"
          >
            Press Releases
          </a>
          <a
            class="sc-AxgMl gwITOl"
          >
            Mission
          </a>
          <a
            class="sc-AxgMl gwITOl"
          >
            Strategy
          </a>
          <a
            class="sc-AxgMl gwITOl"
          >
            About
          </a>
        </div>
        <div
          class="col-sm-2 col-4"
        >
          <h5
            class="sc-AxhUy kaKyLb"
          >
            Info
          </h5>
          <a
            class="sc-AxgMl gwITOl"
          >
            Support
          </a>
          <a
            class="sc-AxgMl gwITOl"
          >
            Customer Service
          </a>
          <a
            class="sc-AxgMl gwITOl"
          >
            Get Started Guide
          </a>
        </div>
        <div
          class="col-sm-3 col-12"
        >
          <h5
            class="sc-AxhUy kaKyLb"
          >
            Follow us
          </h5>
          <div
            class="sc-AxheI AYPRB"
          >
            <span
              class="sc-Axmtr mRqjV"
            >
              <svg>
                facebook.svg
              </svg>
            </span>
            <span
              class="sc-Axmtr mRqjV"
            >
              <svg>
                twitter.svg
              </svg>
            </span>
            <span
              class="sc-Axmtr mRqjV"
            >
              <svg>
                linkedin.svg
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-12"
        >
          <hr
            class="sc-AxmLO feTcaa"
          />
        </div>
        <div
          class="col-sm-12"
        >
          <div
            class="sc-fzozJi ifFRQp"
          >
            © 2019 Manual. All rights reserved.
          </div>
        </div>
      </div>
    </footer>
  </main>
</div>
`;
